src/finch/compiled.py

```diff
@@ -0,0 +1,35 @@
+from functools import wraps
+
+from .julia import jl
+from .tensor import Tensor
+
+
+def compiled(func):
+    @wraps(func)
+    def wrapper_func(*args, **kwargs):
+        new_args = []
+        for arg in args:
+            if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):
+                new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))
+            else:
+                new_args.append(arg)
+
+        result = func(*new_args, **kwargs)
+        result_tensor = Tensor(jl.Finch.compute(result._obj))
+
+        return result_tensor
+
+    return wrapper_func
+
+
+def LazyTensor(tensor: Tensor):
```

Comment:
@willow-ahrens 
> I don't really understand this function. I thought this function was supposed to wrap the input in LazyTensor but it seems like it's computing the tensor?

Reply:
Hmm, I think `def LazyTensor` is a misleading name here. In `finch-tensor`, `Tensor` class contains `_obj` field which is a raw Julia object. It can be either `Finch.SwizzleArray` or `Finch.LazyTensor`.

We need a function that wraps `_obj` in `Finch.LazyTensor` to make a lazy tensor.

Therefore `def LazyTensor` accepts `Tensor` and wraps `_obj` inside in `Finch.LazyTensor`, that's all.

Maybe I can rename it to `def lazy`, or add a method `Tensor.to_lazy()`, so it makes more sense.

Reply:
But I agree that currently it's a confusing name.

Reply:
I'm in favor of `def lazy`, that would at least match the julia interface

Reply:
`A.to_lazy()` feels like the Array-api style way to do things, and `lazy(A)` feels like the julia-style way to do it. I suppose we also call `A.compute()` or `compute(A)` depending. I sortof prefer `lazy(A)` and `compute(A)` for a frontend interface.

Reply:
I think for now I also prefer `lazy(A)` and `compute(A)` versions.